src/inference/src/dev/core_impl_ie.cpp

```diff
@@ -231,28 +260,53 @@ std::map<std::string, std::string> ov::CoreImpl::GetSupportedConfig(const std::s
         }
     } catch (ov::Exception&) {
     }
+    // If device properties for the same device are present several times in different format
+    // they are applied with the following priorities:
+    // 1. Properties of the 1st level
+    // 2. Secondary properties with key DEVICE_PROPERTIES_<DEVICE>
+    // 3. Secondary properties with key DEVICE_PROPERTIES
     std::map<std::string, std::string> supportedConfig;
-    for (auto&& key : supportedConfigKeys) {
-        auto itKey = configs.find(key);
-        if (configs.end() != itKey) {
-            supportedConfig[key] = itKey->second;
-        }
-    }
-    for (auto&& config : configs) {
-        auto parsed = parseDeviceNameIntoConfig(config.first);
-        if (deviceName.find(parsed._deviceName) != std::string::npos) {
-            std::stringstream strm(config.second);
+    auto updateSupportedConfig = [&](const std::string& deviceNameToParse, const std::string& string_value) {
+        auto parsed = parseDeviceNameIntoConfig<std::string>(deviceNameToParse);
+        if (ov::isConfigApplicable(deviceName, parsed._deviceName)) {
```

Comment:
For the case like below config passing to `GetSupportedConfig("GPU.1", configs_as_below)`.
`{
     {"PERFORMANCE_HINT", "UNDEFINED"},
     {"DEVICE_PROPERTIES_GPU.0", "PERFORMANCE_HINT UNDEFINED"}
}`
1. and `if (ov::isConfigApplicable("GPU.1", "GPU")) {` will be reached and  then set the config of "GPU.0" to "GPU.1", right?
2. From my understanding, the expected configs under this logic for target device "GPU.1" should be as below, right?
`{
     {"PERFORMANCE_HINT", "UNDEFINED"}
}`

Reply:
I suppose no. Config specified for GPU.0 is not applicable for GPU.1. If you specified `{"DEVICE_PROPERTIES_GPU", "PERFORMANCE_HINT UNDEFINED"}`, then it will be applied (because GPU is wider name than GPU.1)

You can see examples how `isConfigApplicable` works in unit tests.

Reply:
Yes. as you mentioned, `ov::isConfigApplicable("GPU.0", "GPU.1")` will return `false` in the unit tests. For below case

`auto expectedConfig = core.GetSupportedConfig("GPU.1",
{ {"NUM_STREAMS", "4"}, {"DEVICE_PROPERTIES_GPU.0", "PERFORMANCE_HINT LATENCY"} });`

As my understanding, the `expectedConfig` for "GPU.1" will be `{ {"PERFORMANCE_HINT", "LATENCY"}, {"NUM_STREAMS", "4"}}`, instead of expected configs like `{{"NUM_STREAMS", "4"}}`, right?

Reply:
Maybe we should pass the device name with id, instead of `parsed._deviceName`, to function `ov::isConfigApplicable(deviceName, parsed._deviceName)`.

Reply:
the proper expected config is `{{"NUM_STREAMS", "4"}}`

Please, see this PR https://github.com/openvinotoolkit/openvino/pull/16296/files#diff-8d13c3a18a5f54a062387b01b0e2bb6dedc35c42e73c752ae00b73ee4c60f0fbR114 where those lines are changed to as you suggested